cas-server-support-jpa-ticket-registry/src/main/java/org/jasig/cas/ticket/registry/JpaTicketRegistry.java

```diff
@@ -121,7 +125,8 @@ private Ticket getRawTicket(final String ticketId) {
             if (ticketId.startsWith(TicketGrantingTicket.PREFIX)
                     || ticketId.startsWith(ProxyGrantingTicket.PROXY_GRANTING_TICKET_PREFIX)) {
                 // There is no need to distinguish between TGTs and PGTs since PGTs inherit from TGTs
-                return entityManager.find(TicketGrantingTicketImpl.class, ticketId);
+                return entityManager.find(TicketGrantingTicketImpl.class, ticketId,
+                        lockTgt ? LockModeType.PESSIMISTIC_WRITE : null);
```

Comment:
I was thinking that as a possibility as well, but I do think it's overkill for now. With how I named the property we can always create a different one for the lock type and deprecate the enabled flag if the need actually arises (since this would be replaced by an empty lock type)

Part of the issue comes that the majority of the time the TGT is loaded it is going to be written to during the transaction. But the writes occur a good amount of time after the TGT is loaded but still during the same transaction.

`OPTIMISTIC_READ` and `OPTIMISTIC_WRITE` lock types would require a `version` attribute for the TGT and still presents failures as a deadlock instead of waiting for other transactions to finish, so this type gains nothing over no lock type.
`PESSIMISTIC_READ` locks the TGT with a shared lock which almost always needs to be upgraded to an exclusive write lock later (because we almost always write to the TGT after loading it). This leads to still getting deadlocks because multiple threads would have read the state of the TGT before any of them claim they actually need to write to it.

The only scenario that disabling locks doesn't lead to deadlocks or data inconsistency is when there is no request concurrency within a single TGT/PGT hierarchy, which makes sense in certain deployment/usage scenarios.